icu4c/source/common/bytestriebuilder.cpp

```diff
@@ -485,10 +485,10 @@ BytesTrieBuilder::writeDeltaTo(int32_t jumpTarget) {
             length=2;
         } else {
             if(i<=0xffffff) {
-                intBytes[0]=(char)BytesTrie::kFourByteDeltaLead;
+                intBytes[0]=0xfeu;
```

Comment:
Hmm. I'm not sure we want to go from using defined constants to using magic numbers.
What happens when somebody changes kFourByteDeltaLead in the future? This line is now broken.